android-app/app/src/main/java/de/dhbw/ka/se/fibo/CreateAccountActivity.java

```diff
@@ -55,9 +97,88 @@ private void initializeButton() {
     }
 
     private void registerButtonClicked() {
-        Intent i = new Intent(CreateAccountActivity.this,
+        i = new Intent(CreateAccountActivity.this,
                 MainActivity.class);
-        startActivity(i);
-        finish();
+        create_account_email = findViewById(R.id.create_account_email);
+        String email = create_account_email.getText().toString();
+        passwordField = findViewById(R.id.create_account_password_layer);
+        emailField = findViewById(R.id.create_account_email_layer);
+        create_account_password = findViewById(R.id.create_account_password);
+        String password = create_account_password.getText().toString();
+        createUser(email, password);
+
+    }
+
+    private void createUser(String email, String password) {
+        String url = "http://10.0.2.2:8000/users/register/";
+
+        RequestQueue queue = Volley.newRequestQueue(this);
+
+        System.out.println("Email: "+email);
+        System.out.println("Passwort: "+password);
+
+
+        if (TextUtils.isEmpty(password)&&TextUtils.isEmpty(email)){
+
+            passwordField.setError(getString(R.string.password_field));
+            passwordField.setErrorTextColor(ColorStateList.valueOf(Color.RED));
+            emailField.setError(getString(R.string.email_field));
+            emailField.setErrorTextColor(ColorStateList.valueOf(Color.RED));
+
+        } else if (TextUtils.isEmpty(password)) {
+            passwordField.setError(getString(R.string.password_field));
+            passwordField.setErrorTextColor(ColorStateList.valueOf(Color.RED));
+            emailField.setError(null);
+        } else if (TextUtils.isEmpty(email)) {
+            emailField.setError(getString(R.string.email_field));
+            emailField.setErrorTextColor(ColorStateList.valueOf(Color.RED));
+            passwordField.setError(null);
```

Comment:
I'd like to see this out of the current method and, when we are already at it, simplify this if-else-if-else-if-else.

We could do it like this instead:
1. First, is that we validate the input using a new helper method. In this new method, we run the following steps:
     1. Set `isValid` to true
     2. Reset the errors for `emailField` as well as `passwordField`
     3. Check whether the `emailField` is empty, and if so, run the following steps:
         1.  mark the error with `setError` and `setErrorTextColor`
         2. Do a logical AND operation on `isValid` with false: `isValid &= false`
     4. Check whether the `passwordField` is empty, and if so, run the following steps:
         1. mark the error with `setError` and `setErrorTextColor`
         2. Do a logical AND operation on `isValid` with false: `isValid &= false`
     5. Return `isValid` 
2. If and only if `isValid` is true, run the steps that are under the highlighted code block (i.e. send the request and so forth).

I think this would clean up the code and reduce complexity.

Reply:
This will also help us to write unit tests testing this behavior.

Reply:
The problem is that I dont know how to show a Toast in a activity. Yesterday I had the same problem with the password and errorfield. I asked Jonas for help, but he couldn't helped me either. Maybe you have an idea.

Reply:
what!!!!

Reply:
I have the error. I selected the Pixel 2 for testing. It works on the pixel 4

Reply:
discord?